### PR TITLE
Plasma Mobile Gear: 21.05 -> 21.06 -> 21.07 -> 21.08

### PR DESCRIPTION
**Pull request by samueldr**
#### Motivation for this change

 - https://www.plasma-mobile.org/2021/06/10/plasma-mobile-update-june/
 - https://www.plasma-mobile.org/info/plasma-mobile-gear-21-06/
 - https://www.plasma-mobile.org/2021/07/20/plasma-mobile-gear-21-07/
 - https://www.plasma-mobile.org/2021/08/31/plasma-mobile-gear-21-08/

#### Issues

**:heavy_check_mark:  YES ready for review now**

 - kweather **will segfault under X11**, but runs just fine under Plasma Mobile. ¯\\\_(ツ)\_/¯

#### Relevant people

cc @SCOTT-HAMILTON (for keysmith upgrade)
cc @dotlambda (for angelfish shuffling around)

#### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I ran all the apps under X11, and under a Plasma Mobile VM.

### CODE REVIEW

**Comment by samueldr** [comment]
As expected:

 - https://www.plasma-mobile.org/2021/07/20/plasma-mobile-gear-21-07/

File: pkgs/applications/plasma-mobile/koko.nix [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    kirigami2
+    knotifications
+    kpurpose
+    kquickimageedit
+    qtgraphicaleffects
+    qtlocation
+    qtquickcontrols2
+  ];
+
+  prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/plasma-mobile/koko.nix:68]: You are sure this should not be post?

File: pkgs/applications/plasma-mobile/koko.nix [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    ln -s ${admin1}     src/admin1CodesASCII.txt
+    ln -s ${admin2}     src/admin2Codes.txt
+    ln -s ${cities1000} src/cities1000.zip
+  '';
+
+  meta = with lib; {
+    description = "Image gallery mobile application";
+    homepage = "https://apps.kde.org/koko/";
+    # LGPL-2.1-only OR LGPL-3.0-only OR LicenseRef-KDE-Accepted-LGPL
+    license = [ licenses.lgpl3Only licenses.lgpl21Only ];
```
Comment (SuperSandro2000) [inline pkgs/applications/plasma-mobile/koko.nix:78]:
```suggestion
    license = with licenses; [ lgpl3Only lgpl21Only ];
```

File: pkgs/applications/kde/kpublictransport.nix [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  mkDerivation, lib,
+  extra-cmake-modules,
+}:
+
+mkDerivation {
+  pname = "kpublictransport";
+  meta = {
+    license = [ lib.licenses.cc0 ];
+    maintainers = [ lib.maintainers.samueldr ];
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/kde/kpublictransport.nix:11]:
```suggestion
  meta = with lib; {
    license = [ licenses.cc0 ];
    maintainers = [ maintainers.samueldr ];
  };
```
Also meta should be last.

**Comment by samueldr** [comment]
@dotlambda review requested for angelfish

@SCOTT-HAMILTON review requested for keysmith

**Comment by samueldr** [comment]
### TODO

 - [ ] Break evaluation of `angelfish`'s cargo dependencies to prevent accidental bad updates

**Pull request merged**